engine/src/test/java/com/redhat/cloud/notifications/exports/ExportEventListenerTest.java

```diff
@@ -0,0 +1,640 @@
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.MediaType;
+
+import javax.enterprise.inject.Any;
+import javax.inject.Inject;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.redhat.cloud.notifications.TestConstants.DEFAULT_ORG_ID;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.EXPORT_CHANNEL;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.FILTER_DATE_FROM;
+import static com.redhat.cloud.notifications.exports.ExportEventListener.FILTER_DATE_TO;
+import static org.awaitility.Awaitility.await;
+import static org.mockserver.model.HttpRequest.request;
+import static org.mockserver.model.HttpResponse.response;
+
+@QuarkusTest
+@QuarkusTestResource(TestLifecycleManager.class)
+public class ExportEventListenerTest {
+
+    private static final LocalDate TODAY = LocalDate.now(ZoneOffset.UTC);
+
+    @ConfigProperty(name = "export-service.psk")
+    String exportServicePsk;
+
+    @InjectMock
+    EventRepository eventRepository;
+
+    @Inject
+    ExportEventListener exportEventListener;
+
+    @Any
+    @Inject
+    InMemoryConnector inMemoryConnector;
+
+    @Inject
+    MicrometerAssertionHelper micrometerAssertionHelper;
+
+    /**
+     * Sets up the routes for the Mock Server.
+     */
+    @BeforeEach
+    void setUpMockServerRoutes() {
+        final ClientAndServer mockServer = MockServerLifecycleManager.getClient();
+
+        mockServer
+            .when(request().withPath(".*/error"))
+            .respond(response().withStatusCode(200));
+
+        mockServer
+            .when(request().withPath(".*/upload"))
+            .respond(response().withStatusCode(200));
+    }
+
+    /**
+     * Clears everything from the Mock Server.
+     */
+    @AfterEach
+    void clearMockServer() {
+        MockServerLifecycleManager.getClient().reset();
+    }
+
+    /**
+     * Tests that when an export request is received with an invalid resource
+     * type, then an error is sent to the export service.
+     */
+    @Test
+    void testInvalidResourceTypeRaisesError() {
+        // Save the counter values to assert the "errors count" change later.
+        this.micrometerAssertionHelper.saveCounterValuesBeforeTest(ExportEventListener.EXPORTS_SERVICE_FAILURES_COUNTER);
```

Comment:
Let's save and assert both `EXPORTS_SERVICE_FAILURES_COUNTER` and `EXPORTS_SERVICE_SUCCESSES_COUNTER` in all tests to make sure that they are only incremented when they should be.

Reply:
Done as well!